src/comments.go

```diff
@@ -17,72 +15,52 @@ type GithubClient struct {
 	Client *github.Client
 }
 
-func DeclareClient(client *GithubClient) {
-	token := os.Getenv("GITHUB_TOKEN")
+// NewClient creates a new GithubClient with the provided token
+func NewClient(token string, client *GithubClient) error {
 	if token == "" {
-		log.Fatal("Unauthorized: No token present")
+		return fmt.Errorf("unauthorized: no token present")
 	}
 	client.Ctx = context.Background()
 	client.Client = github.NewClient(nil).WithAuthToken(token)
+	return nil
 }
 
-func GetUserComments(pull int, org, repo, content, filePath string) {
-	var client GithubClient
-	DeclareClient(&client)
-
+func (client *GithubClient) GetUserComments(pull int, org, repo, content, filePath string) (int64, error) {
 	comments, _, err := client.Client.Issues.ListComments(client.Ctx, org, repo, pull, nil)
 	if err != nil {
-		log.Fatalf("Error: %v\n", err)
-	}
-	commentBody := content
-	if filePath != "" {
-		commentBody = files.ParseFileContent(filePath)
+		return 0, fmt.Errorf("error listing comments: %v", err)
```

Comment:
I think you can also include the PR ID in this error? I know the job of the commenter will be on the PR where this will be run but it would not make us think twice and serve as a confirmation it's editing the expected PR.
```suggestion
		return 0, fmt.Errorf("Error listing comments: %v", err)
```

Reply:
fixed